api/types/network/network.go

```diff
@@ -23,9 +23,10 @@ type IPAMConfig struct {
 
 // EndpointIPAMConfig represents IPAM configurations for the endpoint
 type EndpointIPAMConfig struct {
-	IPv4Address  string   `json:",omitempty"`
-	IPv6Address  string   `json:",omitempty"`
-	LinkLocalIPs []string `json:",omitempty"`
+	IPv4Address  string            `json:",omitempty"`
+	IPv6Address  string            `json:",omitempty"`
+	LinkLocalIPs []string          `json:",omitempty"`
+	IpamOpts     map[string]string `json:"IPAMOpts,omitempty"`
```

Comment:
Given it is a member of `EndpointIPAMConfig`, I'd simply call this `Options`. Also in the json tag.